### PR TITLE
Add offset period aggregation type

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:
[No issue] Required for PSSS indicators (eg https://github.com/beyondessential/tupaia-backlog/issues/1338), in order to calculate values for previous weeks (-1, -2 and -3 weeks) in comparison to the selected week

### CODE REVIEW

File: packages/aggregator/src/aggregationTypes.js [hunk 1]
```diff
@@ -4,28 +4,29 @@
...
 
 export const AGGREGATION_TYPES = {
-  RAW: 'RAW',
-  MOST_RECENT: 'MOST_RECENT',
-  MOST_RECENT_PER_ORG_GROUP: 'MOST_RECENT_PER_ORG_GROUP',
-  SUM: 'SUM',
-  SUM_UNTIL_CURRENT_DAY: 'SUM_UNTIL_CURRENT_DAY',
-  SUM_MOST_RECENT_PER_FACILITY: 'SUM_MOST_RECENT_PER_FACILITY',
-  FINAL_EACH_DAY: 'FINAL_EACH_DAY',
+  COUNT_PER_ORG_GROUP: 'COUNT_PER_ORG_GROUP',
```
Comment (kael89) [inline packages/aggregator/src/aggregationTypes.js:7]: Those were sorted alphabetically here + the new type was added (see commit history)

File: packages/aggregator/src/__tests__/analytics/aggregateAnalytics/aggregations/offsetPeriod.test.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    it.each(testData)('%s', (_, { periodType, previous, next }) => {
+      assertPeriodOffsetIsApplied(previous, { periodType, offset: +1 }, next);
+      assertPeriodOffsetIsApplied(next, { periodType, offset: -1 }, previous);
+    });
+
+    it('month - different total days between months', () => {
+      const periodType = 'month';
+      // This is an edge case where applying a - offset is not a reversible with a + offset
+      assertPeriodOffsetIsApplied('20190531', { periodType, offset: -1 }, '20190430');
+      assertPeriodOffsetIsApplied('20190430', { periodType, offset: +1 }, '20190530');
```
Comment (edmofro) [inline packages/aggregator/src/__tests__/analytics/aggregateAnalytics/aggregations/offsetPeriod.test.js:160]: Oh very interesting!

**Review by edmofro — APPROVED** [review]
Wonderful implementation and tests

One thing to consider: we have a very similar config for specifying the default start and end dates of a dashboard report, but it uses `unit` rather than `periodType`, e.g.
```
"defaultTimePeriod": {"unit": "month", "offset": -1}
```

`periodType` is certainly more in keeping with the rest of the aggregation package, so I'm fine with the difference here, but just wanted to raise it.

**Review by edmofro — APPROVED** [review]

**Comment by kael89** [comment]
> Wonderful implementation and tests
> 
> One thing to consider: we have a very similar config for specifying the default start and end dates of a dashboard report, but it uses `unit` rather than `periodType`, e.g.
> 
> ```
> "defaultTimePeriod": {"unit": "month", "offset": -1}
> ```
> 
> `periodType` is certainly more in keeping with the rest of the aggregation package, so I'm fine with the difference here, but just wanted to raise it.

Thanks @edmofro ! Very good point. I'd be keen on using the same key on both instances, so let's keep this in mind for a potential future renaming

By the way, could you please review the [commit](https://github.com/beyondessential/tupaia/pull/1663/commits/45e36e0e70897dd687cf43fbef98bc5a6ee98aa3) I just pushed (I forgot to include a util introduced in [another PR](https://github.com/beyondessential/tupaia/pull/1657))?

**Pull request merged**